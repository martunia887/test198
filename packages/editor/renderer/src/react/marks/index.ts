import { ComponentType } from 'react';
import { Mark } from 'prosemirror-model';

import Alignment from './alignment';
import Annotation from './annotation';
import Breakout from './breakout';
import Code from './code';
import ConfluenceInlineComment from './confluence-inline-comment';
import Em from './em';
import Indentation from './indentation';
import Link from './link';
import Strike from './strike';
import Strong from './strong';
import Subsup from './subsup';
import TextColor from './textColor';
import Underline from './underline';

export const markToReact: { [key: string]: ComponentType<any> } = {
  code: Code,
  em: Em,
  link: Link,
  strike: Strike,
  strong: Strong,
  subsup: Subsup,
  textColor: TextColor,
  underline: Underline,
  annotation: Annotation,

  // Stage0
  confluenceInlineComment: ConfluenceInlineComment,
  breakout: Breakout,
  alignment: Alignment,
  indentation: Indentation,
};

export const toReact = (mark: Mark): ComponentType<any> => {
  return markToReact[mark.type.name];
};

export {
  Code,
  Em,
  Link,
  Strike,
  Strong,
  Subsup,
  TextColor,
  Underline,
  Breakout,
  Annotation,
};
