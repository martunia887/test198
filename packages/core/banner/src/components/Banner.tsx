import * as React from 'react';
import { Container, Content, Icon, Text, Visibility } from '../styled';

interface Props {
  appearance?: 'warning' | 'error' | 'announcement';
  /** Visual style to be used for the banner */
  /** Content to be shown next to the icon. Typically text content but can contain links. */
  children?: React.ReactNode;
  /** Icon to be shown left of the main content. Typically an Atlaskit [@atlaskit/icon](packages/core/icon) */
  icon?: React.ReactChild;
  /** Defines whether the banner is shown. An animation is used when the value is changed. */
  isOpen?: boolean;
  /** Returns the inner ref of the component. This is exposed so the height can be used in page. */
  innerRef?: (element: HTMLElement) => void;
}

class Banner extends React.Component<Props, { height: number }> {
  state = {
    height: 0,
  };

  static defaultProps = {
    appearance: 'warning',
    isOpen: false,
  };

  containerRef?: HTMLElement;

  getHeight = () => {
    if (this.containerRef) {
      this.setState({ height: this.containerRef.clientHeight });
    }
  };

  innerRef = (ref: HTMLElement) => {
    this.containerRef = ref;
    if (this.props.innerRef) {
      this.props.innerRef(ref);
    }
    this.getHeight();
  };

  render() {
    const { appearance, children, icon, isOpen } = this.props;

    return (
      <Visibility bannerHeight={this.state.height} isOpen={isOpen}>
        <Container
          innerRef={this.innerRef}
          appearance={appearance}
          aria-hidden={!isOpen}
          role="alert"
        >
          <Content appearance={appearance}>
            <Icon>{icon}</Icon>
            <Text appearance={appearance}>{children}</Text>
          </Content>
        </Container>
      </Visibility>
    );
  }
}

export default Banner;
