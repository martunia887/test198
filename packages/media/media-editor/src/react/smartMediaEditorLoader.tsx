import * as React from 'react';

import { ModalSpinner } from '@atlaskit/media-ui';
import { colors } from '@atlaskit/theme';
import {
  SmartMediaEditorOwnProps,
  SmartMediaEditorProps,
} from './smartMediaEditor';
import {
  withMediaClient,
  WithOptionalMediaClientProps,
} from '@atlaskit/media-client';

interface State {
  SmartMediaEditor?: React.ComponentType<SmartMediaEditorProps>;
}

export type Props = SmartMediaEditorOwnProps & WithOptionalMediaClientProps;

export class SmartMediaEditorLoader extends React.PureComponent<
  Props,
  State & { isErrored: boolean }
> {
  static displayName = 'AsyncSmartMediaEditor';
  static SmartMediaEditor?: React.ComponentType<SmartMediaEditorProps>;

  state = {
    // Set state value to equal to current static value of this class.
    SmartMediaEditor: SmartMediaEditorLoader.SmartMediaEditor,
    isErrored: false,
  };

  async componentWillMount() {
    if (!this.state.SmartMediaEditor) {
      try {
        const module = await import(/* webpackChunkName:"@atlaskit-internal_smart-media-editor" */
        './smartMediaEditor');
        SmartMediaEditorLoader.SmartMediaEditor =
          module.SmartMediaEditorWithIntl;
        this.setState({ SmartMediaEditor: module.SmartMediaEditorWithIntl });
      } catch (e) {
        // tslint:disable-next-line:no-console
        console.error(e);
        this.setState({ isErrored: true });
      }
    }
  }

  render() {
    const { isErrored } = this.state;
    const { mediaClient } = this.props;

    if (isErrored) {
      return null;
    }
    if (!this.state.SmartMediaEditor || !mediaClient) {
      return (
        <ModalSpinner blankedColor={colors.N700A} invertSpinnerColor={true} />
      );
    }

    return (
      <this.state.SmartMediaEditor {...this.props} mediaClient={mediaClient} />
    );
  }
}

export default withMediaClient(SmartMediaEditorLoader);
