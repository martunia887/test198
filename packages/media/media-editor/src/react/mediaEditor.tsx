import * as React from 'react';
import Spinner from '@atlaskit/spinner';
import {
  MediaEditorContainer,
  OutputArea,
  DrawingCanvas,
  HiddenTextArea,
  HiddenTextHelperDiv,
  SupplementaryCanvas,
  SpinnerWrapper,
  SketchBackground,
} from './styled';
import { Engine } from '../engine/engine';
import {
  ColorWithAlpha,
  Dimensions,
  ExportedImage,
  ShapeParameters,
  TextDirection,
  Tool,
} from '../common';
import { colorSame, colorWithAlphaSame, dimensionsSame } from '../util';

import {
  DefaultDrawingArea,
  OutputSize,
} from '../engine/components/drawingArea';
import {
  DefaultImageProvider,
  urlImageLoader,
} from '../engine/components/imageProvider';
import { DefaultMouseInput } from '../engine/components/mouseInput';
import { DefaultToolbar } from '../engine/components/toolbar';
import { DefaultKeyboardInput } from '../engine/components/keyboardInput';
import { DefaultImageReceiver } from '../engine/components/imageReceiver';
import { DefaultShapeDeleter } from '../engine/components/shapeDeleter';

export const DEFAULT_WIDTH = 845;
export const DEFAULT_HEIGHT = 530;
export const TOOLBAR_HEIGHT = 64;
export const DEFAULT_RATIO = DEFAULT_HEIGHT / DEFAULT_WIDTH;

export type ImageGetter = (format?: string) => ExportedImage;

export interface LoadParameters {
  imageGetter: ImageGetter;
}

export type LoadHandler = (
  imageUrl: string,
  loadParameters: LoadParameters,
) => void;
export type ErrorHandler = (imageUrl: string, error: Error) => void;
export type ShapeParametersChangedHandler = (
  parameters: ShapeParameters,
) => void;

export interface MediaEditorProps {
  imageUrl: string;
  dimensions: Dimensions;
  screenScaleFactor?: number;
  backgroundColor: ColorWithAlpha;
  shapeParameters: ShapeParameters;
  tool: Tool;

  onLoad: LoadHandler;
  onError: ErrorHandler;
  onShapeParametersChanged: ShapeParametersChangedHandler;
  onAnyEdit?: () => void;
  isEmptySketch?: boolean;
}

export interface MediaEditorState {
  isImageLoaded: boolean;
}

const defaultTextDirection = 'ltr';

export class MediaEditor extends React.Component<
  MediaEditorProps,
  MediaEditorState
> {
  private isUnmounted: boolean;

  // DOM elements that we need to create the engine
  private outputArea!: HTMLDivElement;
  private canvas!: HTMLCanvasElement;
  private supplementaryCanvas!: HTMLCanvasElement;
  private hiddenTextArea!: HTMLTextAreaElement;
  private hiddenTextHelperDiv!: HTMLDivElement;

  // Engine and its components
  private drawingArea?: DefaultDrawingArea;
  private toolbar?: DefaultToolbar;
  private engine?: Engine;

  constructor(props: MediaEditorProps) {
    super(props);
    this.isUnmounted = false;
    this.state = {
      isImageLoaded: false,
    };
  }

  componentDidMount() {
    this.loadEngine();
  }

  componentDidUpdate(prevProps: MediaEditorProps) {
    if (!this.engine) {
      return;
    }

    const currProps = this.props;

    if (currProps.imageUrl !== prevProps.imageUrl) {
      this.unloadEngine();
      this.loadEngine();
    }

    if (
      this.drawingArea &&
      (!dimensionsSame(currProps.dimensions, prevProps.dimensions) ||
        currProps.screenScaleFactor !== prevProps.screenScaleFactor)
    ) {
      this.drawingArea.setSize(MediaEditor.toOutputSize(currProps));
    }

    if (
      !colorWithAlphaSame(currProps.backgroundColor, prevProps.backgroundColor)
    ) {
      // TODO inform the core about the new background color
      // https://jira.atlassian.com/browse/FIL-3996
    }

    const {
      color: currColor,
      lineWidth: currLineWidth,
      addShadow: currAddShadow,
    } = currProps.shapeParameters;
    const {
      color: prevColor,
      lineWidth: prevLineWidth,
      addShadow: prevAddShadow,
    } = prevProps.shapeParameters;
    if (this.toolbar) {
      if (!colorSame(currColor, prevColor)) {
        this.toolbar.setColor(currColor);
      }
      if (currLineWidth !== prevLineWidth) {
        this.toolbar.setLineWidth(currLineWidth);
      }
      if (currAddShadow !== prevAddShadow) {
        this.toolbar.setAddShadow(currAddShadow);
      }
      if (currProps.tool !== prevProps.tool) {
        this.toolbar.setTool(currProps.tool);
      }
    }
  }

  componentWillUnmount() {
    this.isUnmounted = true;
    this.unloadEngine();
  }

  private handleOutputAreaInnerRef = (outputArea: HTMLDivElement) => {
    this.outputArea = outputArea;
  };
  private handleSupplementaryCanvasInnerRef = (canvas: HTMLCanvasElement) => {
    this.supplementaryCanvas = canvas;
  };
  private handleHiddenTextAreaInnerRef = (textArea: HTMLTextAreaElement) => {
    this.hiddenTextArea = textArea;
  };
  private handleHiddenTextHelperDivInnerRef = (div: HTMLDivElement) => {
    this.hiddenTextHelperDiv = div;
  };
  private handleDrawingCanvasInnerRef = (canvas: HTMLCanvasElement) => {
    this.canvas = canvas;
  };

  private renderSpinner = () => (
    <SpinnerWrapper>
      <Spinner size="large" invertColor={true} />
    </SpinnerWrapper>
  );

  private get sketchDimensions() {
    const { dimensions } = this.props;
    const sketchDimensions = {
      width: DEFAULT_WIDTH,
      height: DEFAULT_HEIGHT - TOOLBAR_HEIGHT,
    };
    if (dimensions.height < sketchDimensions.height) {
      const extraPadding = 10;
      sketchDimensions.width =
        (dimensions.height - extraPadding) / DEFAULT_RATIO;
      sketchDimensions.height = dimensions.height - extraPadding;
    }
    if (dimensions.width < sketchDimensions.width) {
      const extraPadding = 20;
      sketchDimensions.height =
        (dimensions.width - extraPadding) * DEFAULT_RATIO;
      sketchDimensions.width = dimensions.width - extraPadding;
    }
    return sketchDimensions;
  }

  render() {
    const { isImageLoaded } = this.state;
    const { dimensions, isEmptySketch } = this.props;

    return (
      <MediaEditorContainer style={dimensions}>
        {!isImageLoaded ? this.renderSpinner() : null}
        <OutputArea innerRef={this.handleOutputAreaInnerRef} style={dimensions}>
          <SupplementaryCanvas
            innerRef={this.handleSupplementaryCanvasInnerRef}
          />

          <HiddenTextArea
            autoComplete={'off'}
            innerRef={this.handleHiddenTextAreaInnerRef}
          />

          <HiddenTextHelperDiv
            innerRef={this.handleHiddenTextHelperDivInnerRef}
          />
          {isEmptySketch ? (
            <SketchBackground style={this.sketchDimensions} />
          ) : null}
          <DrawingCanvas
            onClick={this.onCanvasClick}
            innerRef={this.handleDrawingCanvasInnerRef}
            style={dimensions}
          />
        </OutputArea>
      </MediaEditorContainer>
    );
  }

  private onCanvasClick = () => {
    const { onAnyEdit } = this.props;
    if (onAnyEdit) {
      onAnyEdit();
    }
  };

  private loadEngine(): void {
    const { imageUrl, isEmptySketch } = this.props;

    DefaultImageProvider.create(
      () => urlImageLoader(imageUrl),
      this.supplementaryCanvas,
    )
      .then(imageProvider => {
        // We must not create the engine if the component was unmounted or if the image was changed
        if (this.isUnmounted || imageUrl !== this.props.imageUrl) {
          return;
        }
        this.setState({ isImageLoaded: true });

        if (isEmptySketch) {
          const sketchDimensions = this.sketchDimensions;
          imageProvider.backImage.width = sketchDimensions.width;
          imageProvider.backImage.height = sketchDimensions.height;
        }

        // Creating components for the engine
        const outputSize = MediaEditor.toOutputSize(this.props);
        const { backgroundColor } = this.props;
        this.drawingArea = new DefaultDrawingArea(
          this.canvas,
          outputSize,
          backgroundColor,
        );

        const mouseInput = new DefaultMouseInput(this.outputArea);
        this.toolbar = new DefaultToolbar(params =>
          this.props.onShapeParametersChanged(params),
        );
        const keyboardInput = new DefaultKeyboardInput(
          this.hiddenTextArea,
          this.supplementaryCanvas,
          this.hiddenTextHelperDiv,
        );
        const imageReceiver = new DefaultImageReceiver(
          this.supplementaryCanvas,
        );
        const shapeDeleter = new DefaultShapeDeleter(this.hiddenTextArea);

        // Creating the engine
        const { shapeParameters, tool: initialTool } = this.props;
        const textDirection =
          (window.getComputedStyle(this.outputArea)
            .direction as TextDirection) || defaultTextDirection;

        const config = {
          // tslint:disable-next-line:no-console
          onCoreError: (message: string) => {
            // tslint:disable-next-line
            console.error(message);
          },
          shapeParameters,
          initialTool,
          textDirection,
          drawingArea: this.drawingArea,
          imageProvider,
          mouseInput,
          toolbar: this.toolbar,
          keyboardInput,
          imageReceiver,
          shapeDeleter,
        };

        this.engine = new Engine(config);
        const loadParameters: LoadParameters = {
          imageGetter: (format?: string) => this.engine!.getBase64Image(format),
        };

        this.props.onLoad(imageUrl, loadParameters);
      })
      .catch(error => this.props.onError(imageUrl, error));
  }

  private unloadEngine(): void {
    if (this.engine) {
      this.engine.unload();
      delete this.engine;
      this.setState({ isImageLoaded: false });
    }
  }

  private static toOutputSize(props: MediaEditorProps): OutputSize {
    const { dimensions } = props;
    const screenScaleFactor =
      props.screenScaleFactor || MediaEditor.screenScaleFactor;

    return {
      width: dimensions.width * screenScaleFactor,
      height: dimensions.height * screenScaleFactor,
      screenScaleFactor,
    };
  }

  private static get screenScaleFactor(): number {
    return window.devicePixelRatio || 1;
  }
}

export default MediaEditor;
