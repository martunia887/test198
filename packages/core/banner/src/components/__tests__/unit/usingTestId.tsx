import { mount } from 'enzyme';
import { render } from '@testing-library/react';
import React from 'react';
import cases from 'jest-in-case';
import Banner from '../../Banner';

describe('Using enzyme', () => {
  test('It should not generate data-testid', () => {
    const wrapper = mount(
      <Banner isOpen>
        Your license is about to expire. Please renew your license within the
        next week.
      </Banner>,
    );
    expect(wrapper).toBeDefined();
    expect(wrapper.prop('data-testid')).toBeUndefined();
  });

  test('Banner snapshot should be same with data-testid', () => {
    const wrapper = mount(
      <Banner isOpen testId="the-banner">
        Your license is about to expire. Please renew your license within the
        next week.
      </Banner>,
    );
    expect(wrapper.find('[data-testid="the-banner"]')).toMatchSnapshot();
  });

  describe('Banner with different data-testid', () => {
    cases(
      'should be generated',
      ({ key }: { key: string }) => {
        const wrapper = mount(
          <Banner isOpen testId={key}>
            Your license is about to expire. Please renew your license within
            the next week.
          </Banner>,
        );
        expect(wrapper.find(`[data-testid='${key}']`)).toBeTruthy();
      },
      [
        { key: 'josefGiTan' },
        { key: 'ZZZZŹŽ;;;;' },
        { key: '@3$&&&&Helooo' },
        { key: '126^^^' },
        { key: 123 },
      ],
    );
  });
});

describe('Using react-test-library', () => {
  describe('Banner should be found by data-testid', () => {
    test('Using getByTestId()', async () => {
      const testId = 'the-banner';
      const { getByTestId } = render(
        <Banner isOpen testId="the-banner">
          Your license is about to expire. Please renew your license within the
          next week.
        </Banner>,
      );

      expect(getByTestId(testId)).toBeTruthy();
    });
  });
});
