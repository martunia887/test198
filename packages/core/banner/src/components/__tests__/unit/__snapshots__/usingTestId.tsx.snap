// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using enzyme Banner snapshot should be same with data-testid 1`] = `
Array [
  .c0 {
  max-height: 52px;
  overflow: visible;
  background-color: #FFAB00;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFAB00;
  color: #253858;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  fill: #FFAB00;
  font-weight: 500;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 12px;
  text-align: center;
  margin: auto;
  -webkit-transition: color 0.25s ease-in-out;
  transition: color 0.25s ease-in-out;
}

.c1 a,
.c1 a:visited,
.c1 a:hover,
.c1 a:active,
.c1 a:focus {
  color: #253858;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c3 {
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  padding: 4px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

<styled.div
    appearance="warning"
    aria-hidden={false}
    data-testid="the-banner"
    innerRef={[Function]}
    role="alert"
  >
    <div
      aria-hidden={false}
      className="c0"
      data-testid="the-banner"
      role="alert"
    >
      <styled.div
        appearance="warning"
      >
        <div
          className="c1"
        >
          <styled.span>
            <span
              className="c2"
            />
          </styled.span>
          <styled.span
            appearance="warning"
          >
            <span
              className="c3"
            >
              Your license is about to expire. Please renew your license within the next week.
            </span>
          </styled.span>
        </div>
      </styled.div>
    </div>
  </styled.div>,
  .c0 {
  max-height: 52px;
  overflow: visible;
  background-color: #FFAB00;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFAB00;
  color: #253858;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  fill: #FFAB00;
  font-weight: 500;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 12px;
  text-align: center;
  margin: auto;
  -webkit-transition: color 0.25s ease-in-out;
  transition: color 0.25s ease-in-out;
}

.c1 a,
.c1 a:visited,
.c1 a:hover,
.c1 a:active,
.c1 a:focus {
  color: #253858;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c3 {
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  padding: 4px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

<div
    aria-hidden={false}
    className="c0"
    data-testid="the-banner"
    role="alert"
  >
    <styled.div
      appearance="warning"
    >
      <div
        className="c1"
      >
        <styled.span>
          <span
            className="c2"
          />
        </styled.span>
        <styled.span
          appearance="warning"
        >
          <span
            className="c3"
          >
            Your license is about to expire. Please renew your license within the next week.
          </span>
        </styled.span>
      </div>
    </styled.div>
  </div>,
]
`;
